GPU doesn't get recognised with docker image

I can successfully run all tests on CPU but for some reason I can't get the GPU to work on a DGX-1. pytorch's official image works but allennlp one doesn't:

```
$ nvidia-docker run --rm -ti --ipc=host pytorch/pytorch:latest
root@5a84fdfadea1:/workspace# python -c "import torch;print(torch.cuda.is_available())"
True

$ nvidia-docker run -p 8000:8000 -it --rm allennlp/allennlp
root@5fc4428edb93:/stage# python -c "import torch;print(torch.cuda.is_available())"
False
```

Has anyone an idea about what might be going on?

This is a regression.

```
$ sudo docker run -p 8000:8000 -it --rm allennlp/allennlp
root@1650a10e044f:/stage# python -m allennlp.run evaluate \
>     --archive_file https://s3-us-west-2.amazonaws.com/allennlp/models/bidaf-model-2017.09.15-charpad.tar.gz \
>     --evaluation_data_file https://s3-us-west-2.amazonaws.com/allennlp/datasets/squad/squad-dev-v1.1.json --cuda-device 0
/usr/local/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
2018-01-18 16:18:12,617 - WARNING - allennlp.commands - Argument name --archive_file is deprecated (and will likely go away at some point), please use --archive-file instead
2018-01-18 16:18:12,617 - WARNING - allennlp.commands - Argument name --evaluation_data_file is deprecated (and will likely go away at some point), please use --evaluation-data-file instead
2018-01-18 16:18:12,658 - INFO - allennlp.common.file_utils - https://s3-us-west-2.amazonaws.com/allennlp/models/bidaf-model-2017.09.15-charpad.tar.gz not found in cache, downloading to /tmp/tmpeg2457tq
100%|████████████████████████████████████████████████████████████████████████████████████████████████████| 46175392/46175392 [00:01<00:00, 28169374.28B/s]
2018-01-18 16:18:14,444 - INFO - allennlp.common.file_utils - copying /tmp/tmpeg2457tq to cache at /root/.allennlp/datasets/aHR0cHM6Ly9zMy11cy13ZXN0LTIuYW1hem9uYXdzLmNvbS9hbGxlbm5scC9tb2RlbHMvYmlkYWYtbW9kZWwtMjAxNy4wOS4xNS1jaGFycGFkLnRhci5neg==.bc6cc08ed491c6ac831bf363f7312fc1-6
2018-01-18 16:18:14,494 - INFO - allennlp.common.file_utils - removing temp file /tmp/tmpeg2457tq
2018-01-18 16:18:14,495 - INFO - allennlp.models.archival - extracting archive file /root/.allennlp/datasets/aHR0cHM6Ly9zMy11cy13ZXN0LTIuYW1hem9uYXdzLmNvbS9hbGxlbm5scC9tb2RlbHMvYmlkYWYtbW9kZWwtMjAxNy4wOS4xNS1jaGFycGFkLnRhci5neg==.bc6cc08ed491c6ac831bf363f7312fc1-6 to temp dir /tmp/tmp6_7otbk0
2018-01-18 16:18:15,089 - INFO - allennlp.data.vocabulary - Loading token dictionary from /tmp/tmp6_7otbk0/vocabulary.
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/stage/allennlp/run.py", line 18, in <module>
    main(prog="python -m allennlp.run")
  File "/stage/allennlp/commands/__init__.py", line 77, in main
    args.func(args)
  File "/stage/allennlp/commands/evaluate.py", line 103, in evaluate_from_args
    archive = load_archive(args.archive_file, args.cuda_device, args.overrides)
  File "/stage/allennlp/models/archival.py", line 133, in load_archive
    cuda_device=cuda_device)
  File "/stage/allennlp/models/model.py", line 227, in load
    model_state = torch.load(weights_file, map_location=util.device_mapping(cuda_device))
  File "/usr/local/lib/python3.6/site-packages/torch/serialization.py", line 261, in load
    return _load(f, map_location, pickle_module)
  File "/usr/local/lib/python3.6/site-packages/torch/serialization.py", line 409, in _load
    result = unpickler.load()
  File "/usr/local/lib/python3.6/site-packages/torch/serialization.py", line 380, in persistent_load
    data_type(size), location)
  File "/usr/local/lib/python3.6/site-packages/torch/serialization.py", line 278, in restore_location
    result = map_location(storage, location)
  File "/stage/allennlp/nn/util.py", line 460, in inner_device_mapping
    return storage.cuda(cuda_device)
  File "/usr/local/lib/python3.6/site-packages/torch/_utils.py", line 61, in _cuda
    with torch.cuda.device(device):
  File "/usr/local/lib/python3.6/site-packages/torch/cuda/__init__.py", line 186, in __enter__
    _lazy_init()
  File "/usr/local/lib/python3.6/site-packages/torch/cuda/__init__.py", line 120, in _lazy_init
    _check_driver()
  File "/usr/local/lib/python3.6/site-packages/torch/cuda/__init__.py", line 62, in _check_driver
    http://www.nvidia.com/Download/index.aspx""")
AssertionError: 
Found no NVIDIA driver on your system. Please check that you
have an NVIDIA GPU and installed a driver from
http://www.nvidia.com/Download/index.aspx
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -2,9 +2,12 @@ FROM python:3.6.3-jessie
 
 ENV LC_ALL=C.UTF-8
 ENV LANG=C.UTF-8
-ENV PATH /usr/local/nvidia/bin/:$PATH
 
-ENV LD_LIBRARY_PATH /usr/local/nvidia/lib64
+ENV PATH /usr/local/nvidia/bin/:$PATH
+ENV LD_LIBRARY_PATH /usr/local/nvidia/lib:/usr/local/nvidia/lib64
+
+# Tell nvidia-docker that we need the driver, which gets mounted at /usr/local/nvidia.
+LABEL com.nvidia.volumes.needed="nvidia_driver"
 
 WORKDIR /stage
 

